src/main/lws/lws-service.js

```diff
@@ -7,13 +7,256 @@ import { IdAttribute } from '../identity/id-attribute';
+	'chrome-extension://knldjmfmopnpolahpmmgbagdohdnhkik',
+	'chrome-extension://pfdhoblngboilpfeibdedpjgfnlcodoo'
+];
 export const WS_IP_WHITELIST = ['127.0.0.1', '::1'];
-export const WS_PORT = process.env.LWS_WS_PORT || 8898;
+export let WS_PORT = process.env.LWS_WS_PORT || 8898;
 
 const log = new Logger('LWSService');
 
+const userDataPath = common.getUserDataPath();
+
+function init() {
+	return new Promise((resolve, reject) => {
+		try {
+			let osxConfig = {
+				lwsPath: userDataPath + '/lws/',
+				lwsKeyPath: userDataPath + '/lws/keys',
+				reqFile: userDataPath + '/lws/keys/lws_cert.pem',
+				rsaFile: userDataPath + '/lws/keys/lws_key.pem',
+				keyTempFile: userDataPath + '/lws/keys/keytemp.pem',
+				certgen: [
+					{
+						cmd:
+							'openssl req \
+							-new \
+							-newkey rsa:2048 \
+							-days 365 \
+							-nodes \
+							-x509 \
+							-subj "/C=NV/ST=SK/L=Nevis/O=selfkey/CN=localhost" \
+							-extensions EXT \
+							-config <( printf "[dn]\nCN=localhost\n[req]\ndistinguished_name = dn\n[EXT]\nsubjectAltName=DNS:localhost\nkeyUsage=digitalSignature\nextendedKeyUsage=serverAuth") \
+							-keyout "' +
+							keyTempFile +
+							'" \
+							-out "' +
+							reqFile +
+							'"',
+						options: {
+							shell: '/bin/bash'
+						},
+						type: 'child'
+					},
+					{
+						cmd: 'openssl rsa \
+							-in "' + keyTempFile + '" \
+							-out "' + rsaFile + '"',
+						options: {
+							shell: '/bin/bash'
+						},
+						type: 'child'
+					},
+					{
+						cmd:
+							'security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain "' +
+							reqFile +
+							'"',
+						options: {
+							name:
+								'SelfKey needs to install a security certifcate to encrypt data and'
+						},
+						type: 'sudo'
+					}
+				]
+			};
+			let linConfig = {
+				lwsPath: userDataPath + '/lws/',
+				lwsKeyPath: userDataPath + '/lws/keys',
+				reqFile: userDataPath + '/lws/keys/lws_cert.pem',
+				rsaFile: userDataPath + '/lws/keys/lws_key.pem',
+				keyTempFile: userDataPath + '/lws/keys/keytemp.pem',
+				certgen: [
+					{
+						cmd:
+							'openssl req \
+							-new \
+							-newkey rsa:2048 \
+							-days 365 \
+							-nodes \
+							-x509 \
+							-subj "/C=NV/ST=SK/L=Nevis/O=selfkey/CN=localhost" \
+							-extensions EXT \
+							-config <( printf "[dn]\nCN=localhost\n[req]\ndistinguished_name = dn\n[EXT]\nsubjectAltName=DNS:localhost\nkeyUsage=digitalSignature\nextendedKeyUsage=serverAuth") \
+							-keyout "' +
+							keyTempFile +
+							'" \
+							-out "' +
+							reqFile +
+							'"',
+						options: {
+							shell: '/bin/bash'
+						},
+						type: 'child'
+					},
+					{
+						cmd: 'openssl rsa \
+							-in "' + keyTempFile + '" \
+							-out "' + rsaFile + '"',
+						options: {
+							shell: '/bin/bash'
+						},
+						type: 'child'
+					}
+				]
+			};
+			let winConfig = {
+				lwsPath: userDataPath + '\\lws\\',
```

Comment:
@altninja I think if you use path.join it will already handle differences between  OSs so you don't need to repeat these paths.